firebase-database/src/main/java/com/google/firebase/database/Query.java

```diff
@@ -331,7 +334,13 @@ public Query startAfter(boolean value) {
   @NonNull
   public Query startAfter(@Nullable String value, @Nullable String key) {
     Node node =
-        value != null ? new StringNode(value, PriorityUtilities.NullPriority()) : EmptyNode.Empty();
+        value != null
+            ? new StringNode(
+                (params.getIndex().equals(KeyIndex.getInstance()))
+                    ? PushIdGenerator.successor(value)
+                    : value,
+                PriorityUtilities.NullPriority())
+            : EmptyNode.Empty();
```

Comment:
Can we rewrite this to match the style above? It's a bit hard to parse with two nested conditionals.

Reply:
Also - do we not need the validation from https://github.com/firebase/firebase-ios-sdk/pull/7403/files#diff-fa09d2f832ea9e4ec0b8ca482585897a8706c7b6c2a0218250e08ed78c7c3c9aR205 here?

Reply:
I commented on that PR, but this case is covered in `validateQueryEndpoints`: 

https://github.com/firebase/firebase-android-sdk/blob/master/firebase-database/src/main/java/com/google/firebase/database/Query.java#L84-L103